modules/core/field/src/Plugin/Field/FieldFormatter/UrlLinkFormatter.php

```diff
@@ -28,6 +28,7 @@ public function viewElements(FieldItemListInterface $items, $langcode) {
 
     // This is a clone of Drupal core's UriLinkFormatter, but it only renders
     // a link if the URI is a valid URL.
+    /** @var \Drupal\Core\Field\Plugin\Field\FieldType\UriItem[] $items */
```

Comment:
```suggestion
    /** @var \Drupal\Core\Field\Plugin\Field\FieldType\UriItem $item */
```

@mstenta minor detail but `$items` isn't actually an array of `UriItem`. [`FieldItemListInterface`](https://git.drupalcode.org/project/drupal/-/blob/10.5.x/core/lib/Drupal/Core/Field/FieldItemListInterface.php?ref_type=heads#L28) implements `ListInterface` which allows it to be iterated over in the foreach loop and yield each separate field item.

This would be important if we were using other methods from `$items` that are unique to `FieldItemListInterface`. I think we should add the correct type hint on `$item` itself.

Reply:
If for some reason that isn't working we might need to move the type hint to be inside the foreach loop. But I think I have done this before

Reply:
Ah thanks @paul121. Makes sense.

Fixup commit: https://github.com/farmOS/farmOS/pull/943/commits/3ceb4eaf6073b1e70588c34c28941621e860a5e7